url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/agent/KitRobotAgent.java

```diff
@@ -0,0 +1,461 @@
+		numKitsRequested = 0;
+		numKitsToMake = 0;
+		state = KitRobotState.IDLE;
+
+		// Don't assume stand is empty
+		standPositions.put(0, true);
+		standPositions.put(1, false);
+		standPositions.put(2, false);
+	}
+
+	/*
+	 * Messages
+	 */
+
+	@Override
+	public void msgNeedThisManyKits(int total) {
+		print("Received msgNeedThisManyKits with amount " + total);
+		conveyor.msgNeedThisManyKits(total);
+		numKitsToMake = total;
+		numKitsRequested = 0;
+		kitRequested = false;
+		stateChanged();
+	}
+
+	@Override
+	public void msgKitReadyForPickup() {
+		print("Received msgKitReadyForPickup");
+		kitWaitingOnConveyor = true;
+		stateChanged();
+	}
+
+	@Override
+	public void msgNoKitsLeftOnConveyor() {
+		print("Received msgNoKitsLeftOnConveyor");
+		kitWaitingOnConveyor = false;
+		// stateChanged();
+	}
+
+	@Override
+	public void msgHereIsKit(Kit k) {
+		print("Received msgHereIsKit");
+		print("Adding kit " + k.toString());
+		kitRequested = false;
+		MyKit mk = new MyKit(k);
+		myKits.add(mk);
+		stateChanged();
+	}
+
+	@Override
+	public void msgNeedKit(int standLocation) {
+		print("Received msgNeedKit for stand location " + standLocation);
+		standPositions.put(standLocation, true);
+		stateChanged();
+	}
+
+	@Override
+	public void msgMoveKitToInspectionArea(Kit k) {
+		print("Received msgMoveKitToInspectionArea");
+		print(k.toString() + " should be moved.");
+		synchronized (myKits) {
+			// print("Acquiring in MKTIA");
+			print("Mykits size: " + myKits.size());
+			for (MyKit mk : myKits) {
+				if (mk.kit.equals(k)) {
+					mk.KS = KitStatus.MARKED_FOR_INSPECTION;
+					print("Found kit to inspect");
+					break;
+				}
+			}
+		}
+		stateChanged();
+	}
+
+	@Override
+	public void msgKitPassedInspection() {
+		print("Received msgKitPassedInspection");
+		synchronized (myKits) {
+			// print("Acquiring in KPI");
+			for (MyKit mk : myKits) {
+				if (mk.KS == KitStatus.AWAITING_INSPECTION) {
+					mk.KS = KitStatus.PASSED_INSPECTION;
+					break;
+				}
+			}
+		}
+		stateChanged();
+	}
+
+	@Override
+	public void msgKitFailedInspection() {
+		print("Received msgKitFailedInspection");
+		synchronized (myKits) {
+			for (MyKit mk : myKits) {
+				if (mk.KS == KitStatus.AWAITING_INSPECTION) {
+					mk.KS = KitStatus.FAILED_INSPECTION;
+					break;
+				}
+			}
+		}
+		stateChanged();
+	}
+
+	@Override
+	public void msgPlaceKitOnConveyorDone() {
+		print("Received msgPlaceKitOnConveyorDone from graphics");
+		animation.release();
+		// stateChanged();
+	}
+
+	@Override
+	public void msgPlaceKitInInspectionAreaDone() {
+		print("Received msgPlaceKitInInspectionAreaDone from graphics");
+		animation.release();
+		// stateChanged();
+	}
+
+	@Override
+	public void msgPlaceKitOnStandDone() {
+		print("Received msgPlaceKitOnStandDone from graphics");
+		animation.release();
+		// stateChanged();
+	}
+
+	/*
+	 * Scheduler
+	 * 
+	 * @see agent.Agent#pickAndExecuteAnAction()
+	 */
+	@Override
+	public boolean pickAndExecuteAnAction() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 220-287 into a separate method.